pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -186,23 +183,32 @@ func NewDeploymentController(client kube.Client, clusterID cluster.ID, env *mode
 	dc.serviceAccounts.AddEventHandler(parentHandler)
 	dc.clients[gvr.ServiceAccount] = NewUntypedWrapper(dc.serviceAccounts)
 
-	dc.namespaces = kclient.New[*corev1.Namespace](client)
-	dc.namespaces.AddEventHandler(controllers.ObjectHandler(func(o controllers.Object) {
-		// TODO: make this more intelligent, checking if something we care about has changed
-		// requeue this namespace
-		for _, gw := range dc.gateways.List(o.GetName(), klabels.Everything()) {
-			dc.queue.AddObject(gw)
-		}
-	}))
+	dc.gateways = kclient.New[*gateway.Gateway](client)
+	dc.gateways.AddEventHandler(controllers.ObjectHandler(dc.queue.AddObject))
 
-	gateways.AddEventHandler(controllers.ObjectHandler(dc.queue.AddObject))
-	gatewayClasses.AddEventHandler(controllers.ObjectHandler(func(o controllers.Object) {
-		for _, g := range dc.gateways.List(metav1.NamespaceAll, klabels.Everything()) {
-			if string(g.Spec.GatewayClassName) == o.GetName() {
-				dc.queue.AddObject(g)
+	if client.GetMemberRoll() == nil {
+		dc.namespaces = kclient.New[*corev1.Namespace](client)
+		dc.namespaces.AddEventHandler(controllers.ObjectHandler(func(o controllers.Object) {
+			// TODO: make this more intelligent, checking if something we care about has changed
+			// requeue this namespace
+			for _, gw := range dc.gateways.List(o.GetName(), klabels.Everything()) {
+				dc.queue.AddObject(gw)
 			}
-		}
-	}))
+		}))
+		dc.gatewayClasses = kclient.New[*gateway.GatewayClass](client)
+		dc.gatewayClasses.AddEventHandler(controllers.ObjectHandler(func(o controllers.Object) {
+			for _, g := range dc.gateways.List(metav1.NamespaceAll, klabels.Everything()) {
+				if string(g.Spec.GatewayClassName) == o.GetName() {
+					dc.queue.AddObject(g)
+				}
+			}
+		}))
+	}
+
+	if tw != nil {
```

Comment:
Actually, this condition might be merged with the previous one (`GetMemberRoll() == nil`), because `tw` is also created conditionally, but outside the gateway controller, so I didn't know if this is elegant. Any opinion?